File: src/APOGEE.sav/apogee/libapogee-3.0.3234/examples/SimpleExposure.cpp
#include <iostream>
#include <string>
#include <vector>
#include <sstream>
#include <stdint.h>
#include <stdexcept>

#include "apogee/Alta.h"
#include "apogee/FindDeviceUsb.h"
//#include "apogee/FindDeviceEthernet.h"
#include "apogee/CameraInfo.h"
#include "apgSampleCmn.h"

////////////////////////////
// MAIN
int main()
{
	try
	{
        std::cout << "Apogee Alta Simple Exposoure Sample" << std::endl;
        
        // use this for ethernet cameras
        // std::string ioInterface("ethernet");
        // FindDeviceEthernet look4cam;
        // std::string subnet = "192.168.0.255";
        // std::string msg = look4cam.Find( subnet );
		// std::string addr = apgSampleCmn::GetEthernetAddress( msg );
        
        // for this example only looking for only one usb device
        std::string ioInterface("usb");
        FindDeviceUsb look4cam;
        std::string msg = look4cam.Find();
                
        std::string addr = apgSampleCmn::GetUsbAddress( msg );
        uint16_t id = apgSampleCmn::GetID( msg );
        uint16_t frmwrRev = apgSampleCmn::GetFrmwrRev( msg );
        
        //create the camera object and conneting to the camera
		// assuming alta camera for this example, use the Ascent
		// camera object for Ascent cameras.
        Alta cam;
        cam.OpenConnection( ioInterface, addr, frmwrRev, id );
		cam.Init();
		
        // print basic camera info
        std::cout << "Connected to Alta camera " << std::endl;
		apgSampleCmn::printInfo( cam.GetModel(),  cam.GetMaxImgRows(),
								cam.GetMaxImgCols() );
        
        // fetch the default image size
		const uint16_t rows = cam.GetRoiNumRows();
		const uint16_t cols = cam.GetRoiNumCols();
		
		//allocate vector for the image data
		std::vector<uint16_t> img( rows*cols );

		// start 0.1 sec light (shutter will open) expousre
		const double exposeTime = 0.1;
		std::cout << "Starting a " << exposeTime << " sec light exposure" << std::endl;
		cam.StartExposure( exposeTime, true );
		
		// Check camera status to make sure image data is ready
		Apg::Status status = Apg::Status_Flushing;
		while( Apg::Status_ImageReady !=  status )
		{
			status = cam.GetImagingStatus();
			//make sure there isn't an error
			//throw here if there is
			apgSampleCmn::checkStatus( status );
		}

		std::cout << "Retrieving image data from camera" << std::endl;

		cam.GetImage( img );

		std::cout << "Writing image to file" << std::endl;
		
		// Write the test image to an output file (overwrite if it already exists)
		apgSampleCmn::saveImg2File( (char*)&img[0], "simpleExposure.bin",
					 img.size()*sizeof(uint16_t) );
	
        std::cout << "Closing connection to camera " << std::endl;
        cam.CloseConnection();
	}
	catch( std::exception & err )
	{
		std::cout << "std exception what = " << err.what() << std::endl;
		return 1;
	}
	catch( ... )
	{
		std::cout << "Unknown exception caught." << std::endl;
		return 1;
	}

	return 0;
}

File: src/APOGEE.sav/apogee/apps/python/SimpleExposure.py
import numpy
import pylibapogee.pylibapogee as apg
import pylibapogee.pylibapogee_setup as SetupDevice

print "Trying to find and connect with camera"
#look for usb cameras first
devices = SetupDevice.GetUsbDevices()

# no usb cameras, then look for ethernet cameras
if( len(devices) == 0 ):
    devices = SetupDevice.GetEthernetDevices()
    
# exception....no cameras anywhere....
if( len(devices) == 0 ):
    raise RuntimeError( "No devices found on usb or ethernet" )
    
# connect to the first camera
cam = SetupDevice.CreateAndConnectCam( devices[0] )

#print some basic info
print "Imaging rows = %d, columns = %d" % ( cam.GetMaxImgRows(),
								cam.GetMaxImgCols() )

cam.SetCooler( True )

cam.SetImageCount( 1 )

exposeTime = 0.2
print "Starting %f sec light exposure" % (exposeTime) 
cam.StartExposure( exposeTime, True )
                
status = None
while status != apg.Status_ImageReady:
    
    status = cam.GetImagingStatus()	
    if( apg.Status_ConnectionError == status or
        apg.Status_DataError == status or
        apg.Status_PatternError == status ):
        msg = "Run %s: FAILED - error in camera status = %d" % (runStr, status)
        raise RuntimeError( msg )
        
print "Getting image"
data = cam.GetImage()

print "Saving image to file"
imgName = "simpleExpose.bin"
data.tofile( imgName )
					
cam.CloseConnection()